.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,41 @@
+# Standard GPflow pull request
+
+At first, thank you very much for spending time on contributing to GPflow.\
+This template exists to help us to grasp your work without asking very common and questions. The template helps you to get thorough feedback as well.
+
+
+## PR content:
+
+* [Title](#title)
+* [Introduction](#introduction)
+* [Minimal working example](#minimal-working-example)
+
+### Title
+
+Squeeze it. Be creative. There is no necessity for any marks like [BUG], {feature} and cetera. We have github labels for that.
+
+### Description
+
+Describe your contribution by writing down the context and reasons. Make it clear whether it is a **bug fix**, **feature** or **enhancement**. Do not make it very wordy, keep it short and simple.
+
+* Bad example: `This is a bug-fix found at #111.`
+* Bad example: `Add new SuperMagic optimizer.`
+* Good example: `Working on feature #111, we found that parameters of VGP are not instantiated properly. It causes arbitrary failures at training. The bug is fixed by adding necessary parameters to the list of trainables.`
+
+
+### Minimal working example
+
+No matter how good the description is, the best thing you can do for making a reviewer's life much easier is to add a minimal working example (MWE). Ideally, it should be a short code snippet with relevant comments. If a MWE is big (> 50 lines), you should consider [gist](https://gist.github.com) instead.
+
+* If your PR is a *bug* related fix, then your MWE must disclose the bug using synthetic example or "real" one.
```

Comment:
In some cases, people work on the project which they can't share in open source community, I suppose, saying "real" I meant projects which they can share and by synthetic I meant cases where they have to make up a situation, so that the bug is exposed.